internal/store/firestore.go

```diff
@@ -0,0 +1,154 @@
+package firestore
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/firestore"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"pantheon.io/edrt-policy-docs-functions/internal/types"
+)
+
+const hostnameCollectionName = "hostnames"
+const hostnameMetadataCollectionName = "hostnameMetadata"
+const edgeLogicCollectionName = "edgelogic"
+
+type Firestore struct {
+	logger    *logrus.Logger
+	projectId string
+}
+
+func New(logger *logrus.Logger, projectId string) Firestore {
+	return Firestore{
+		logger:    logger,
+		projectId: projectId,
+	}
+}
+
+func (f *Firestore) Read(ctx context.Context) interface{} {
+	client, err := firestore.NewClient(ctx, f.projectId)
```

Comment:
There is no need in creating a new firestore client everytime we need to read the doc. The client can be reused. In your case it can look like:
```go
type Firestore struct {
	logger    *logrus.Logger
	client    *firestore.Client
}


func (f *Firestore) Read(ctx context.Context) interface{} {
	collection := f.client.Collection("foo")
	doc := collection.Doc("bar")
	// ...
}
```

Reply:
Yes, this was in the back of my mind and I meant to fix that. Thanks for calling that out.

Reply:
This has been fixed